### PR TITLE
[17.0][MIG] sale_manual_delivery: Migration to 17.0

### PR DESCRIPTION
**Pull request by astirpe**

### CODE REVIEW

**Comment by youring** [comment]
@astirpe would you cherry pick this one https://github.com/ForgeFlow/sale-workflow/pull/5
 ...to keep the CREATE DELIVERY button shown on partial delivered sale order. Thank you.

**Comment by astirpe** [comment]
@youring done!

**Comment by youring** [comment]
Functional! 
Tested locally, found two warnings on install

> 2024-02-28 15:13:55,177 105178 WARNING v17 odoo.addons.base.models.ir_model: Rule access_manual_delivery_all has no group, this is a deprecated feature. Every access-granting rule should specify a group.
> 2024-02-28 15:13:55,177 105178 WARNING v17 odoo.addons.base.models.ir_model: Rule access_manual_delivery_line_all has no group, this is a deprecated feature. Every access-granting rule should specify a group.

Maybe it's a good time to add default group.

**Comment by astirpe** [comment]
@youring done!

File: sale_manual_delivery/wizard/manual_delivery.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                            <field name="date_planned" />
+                        </group>
+                        <group>
+                            <field name="carrier_id" />
+                            <field name="route_id" groups="stock.group_adv_location" />
+                        </group>
+                    </group>
+                    <field name='line_ids'>
+                        <tree editable="bottom" create="0">
+                            <field name="order_line_id" invisible="1" force_save="1" />
```
Comment (youring) [inline sale_manual_delivery/wizard/manual_delivery.xml]:
```suggestion
                            <field name="order_line_id" column_invisible="1" force_save="1" />
```

**Review by youring — CHANGES_REQUESTED** [review]
Sorry i didn't notice that it was upgraded automatically by views_migration_17 in my local env. 
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#tasks-to-do-in-the-migration

File: sale_manual_delivery/wizard/manual_delivery.xml (hunk 1 above)
Comment (astirpe) [reply sale_manual_delivery/wizard/manual_delivery.xml]: Done. Thanks!

**Comment by youring** [comment]
LGTM. Functional.

**Review by rousseldenis — APPROVED** [review]
Code reiview

**Pull request merged**